### PR TITLE
[merged] pipeline: Resolve context.TODO()

### PR DESCRIPTION
**Pull request by archseer**

### CODE REVIEW

File: core/services/pipeline/scheduler.go [hunk 1]
```diff
@@ -158,11 +158,8 @@ func (s *scheduler) Run() {
 		// pipeline is completely empty
 
 		result := <-s.resultCh
-		// var result TaskRunResult
-		// select {
-		// case result = <-s.resultCh:
-		// case <-s.ctx.Done():
-		// }
+		// TODO: if for some reason the cleanup didn't succeed and we're stuck waiting for reports forever
+		// we should be able to timeout and finish shutting down
```
Comment (samsondav) [inline core/services/pipeline/scheduler.go:162]: What is the TODO exactly?

**Review by samsondav — APPROVED** [review]

**Review by vyzaldysanchez — APPROVED** [review]

**Comment by samsondav** [comment]
@archseer do we want to close this in favour of:

https://github.com/smartcontractkit/chainlink/pull/5384

?

**Comment by archseer** [comment]
@cl-homu r+

**Comment by cl-homu** [comment]
:pushpin: Commit 89b6953afb8a30f8a966b1e1f2075f3796f10b2f has been approved by `archseer`

**Comment by cl-homu** [comment]
:hourglass: Testing commit 89b6953afb8a30f8a966b1e1f2075f3796f10b2f with merge 7e8d4c95a3ff9107ffa74689e2ffd0d77b425240...

**Comment by cl-homu** [comment]
:broken_heart: Test failed - [checks-core](https://github.com/smartcontractkit/chainlink/runs/4164847237?check_suite_focus=true)

**Comment by samsondav** [comment]
@cl-homu retry

**Comment by cl-homu** [comment]
:hourglass: Testing commit 89b6953afb8a30f8a966b1e1f2075f3796f10b2f with merge dacfecdacff012308d3872156248549902dd7903...

**Comment by cl-homu** [comment]
:sunny: Test successful - [checks-core](https://github.com/smartcontractkit/chainlink/runs/4164971776?check_suite_focus=true), [checks-lint](https://github.com/smartcontractkit/chainlink/runs/4164971772?check_suite_focus=true), [checks-npm](https://github.com/smartcontractkit/chainlink/runs/4164971999?check_suite_focus=true), [checks-operator](https://github.com/smartcontractkit/chainlink/runs/4164971923?check_suite_focus=true), [checks-prettier](https://github.com/smartcontractkit/chainlink/runs/4164971988?check_suite_focus=true), [checks-solidity](https://github.com/smartcontractkit/chainlink/runs/4164971869?check_suite_focus=true)
Approved by: archseer
Pushing dacfecdacff012308d3872156248549902dd7903 to develop...

**Pull request closed without merging**